module/registry.mjs

```diff
@@ -421,11 +520,64 @@ class SummonRegistry {
   }
 }
 
+/* -------------------------------------------- */
+/*  Public API                                  */
+/* -------------------------------------------- */
+
+/**
+ * Track the ready status of various registries.
+ * @type {Map<string, boolean>}
+ */
+const RegistryStatus = new class extends Map {
+  /**
+   * Callbacks to call when spell lists are ready.
+   * @type {Function[]}
+   */
+  callbacks = [];
+
+  /* -------------------------------------------- */
+
+  /**
+   * Are all registries ready?
+   * @returns {boolean}
+   */
+  get isReady() {
+    return Array.from(this.values()).every(s => s);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  set(key, value) {
+    super.set(key, value);
+    if ( this.isReady ) {
+      let callback = this.callbacks.shift();
+      while ( callback ) {
+        callback();
+        callback = this.callbacks.shift();
+      }
+    }
+  }
+}();
+
+/* -------------------------------------------- */
+
+/**
+ * Execute a callback when all registries have finished loading.
+ * @param {Function} callback
+ */
+function whenReady(callback) {
```

Comment:
I think providing a `Promise` API would be a bit more ergonomic. It offers a pretty similar interface, but you can also optionally `await` it, e.g.:

```js
// This:
dnd5e.registry.whenReady(() => {
  // ...
});

// Becomes:
dnd5e.registry.whenReady().then(() => {
  // ...
});

// Or:
await dnd5e.registry.whenReady();
// ...
```

```js
const RegistryStatus = new class extends Map {
  constructor() {
    super();
    this.#ready = new Promise(resolve => this.#resolve = resolve);
  }

  #ready;
  #resolve;

  get ready() { return this.#ready };

  set(key, value) {
    super.set(key, value);
    if ( Array.from(this.values()).every(s => s) ) this.#resolve();
  }
}();

export default {
  // [snip]
  whenReady: RegistryStatus.ready
}
```